lib/backend.js

```diff
@@ -661,24 +661,33 @@ Backend.prototype.querySubscribe = function(agent, index, query, options, callba
         'DB does not support subscribe'
       ));
     }
-    backend.pubsub.subscribe(request.channel, function(err, stream) {
-      if (err) return callback(err);
-      if (options.ids) {
-        var queryEmitter = new QueryEmitter(request, stream, options.ids);
-        backend.emit('timing', 'querySubscribe.reconnect', Date.now() - start, request);
-        callback(null, queryEmitter);
-        return;
-      }
-      // Issue query on db to get our initial results
-      backend._query(agent, request, function(err, snapshots, extra) {
-        if (err) {
-          stream.destroy();
-          return callback(err);
+
+    var channels = [request.channel];
+
+    if (request.channels.length) {
+      channels = request.channels;
+    }
+
+    channels.forEach(function(channel) {
```

Comment:
What happens if we have a query listening on multiple channels, and an op that broadcasts on multiple matching channels? Will we get duplicate events being emitted?

Reply:
Yeah that's a good point i would guess yes, but not sure if we can prevent that. 🤔